### DISCUSSION TITLE
Improve 'Save' action in form builder

### DISCUSSION THREAD

**Discussion by happytodev** [filamentphp/filament#3970] [Ideas]
As a user, I wish to have either a floating save button especially when I use the builder or a keyboard shortcut to save like 'ctrl + s' to avoid scroll down the page to find the save button.

**Comment by danharrin** [comment]
Ctrl + S is already a keyboard shortcut, no?

**Reply by happytodev** [reply]
It already works to save a form ?

**Reply by danharrin** [reply]
Yes

**Reply by happytodev** [reply]
Is there something to activate to have it working ? Because it currently doesn't work for me :-(

I see this https://filamentphp.com/docs/2.x/admin/pages/actions#keybindings
But it is for the admin pages.

So how can I activate it in my resources  when I create my forms ?

**Reply by danharrin** [reply]
it should be activated by default, works fine for me, just tested

**Reply by happytodev** [reply]
I'm sorry Dan but for me, not.

I tried with my app without success.

So to be sure, I installed a fresh demo from your Github (https://github.com/filamentphp/demo) and try to edit something in this page : `/blog/posts/1/edit?activeRelationManager=0`

and after try `ctrl+s` or `cmd+s` , nothing happened.

**Reply by danharrin** [reply]
I think that the Select field breaks it as Choices.js captures the keybindings

Try it on this page - https://demo.filamentphp.com/shop/brands/11/edit?activeRelationManager=0

**Reply by happytodev** [reply]
Ok I tried on this field : 

![image](https://user-images.githubusercontent.com/425998/189751114-81e73537-4c21-49ec-9201-c38514cf0ae6.png)

So after modifying text, I use `cmd + s` keys. It works... But ... If I add more text and I try to save nothing happens.

I need to reload page to save one more time

It seems when a save action is fired, it is not possible to fire an another save.

Hope it helps

**Reply by happytodev** [reply]
Maybe it's a bug ?

**Reply by happytodev** [reply]
A little more precision, on my CMS project using TALL and Filament, no save action at all by using keybindings :-(

**Reply by happytodev** [reply]
@danharrin do you reproduce the same ?

**Reply by happytodev** [reply]
@danharrin did you seen my previous messages ?

**Reply by danharrin** [reply]
Yes, but fixing this is not a priority for me right now, just a minor user experience thing. There are lots of other bugs that need my attention, and someone else who has time can look into this if they want.
